### PR TITLE
Remove static functions

### PR DESCRIPTION
**Pull request by jerrywham**
## Removed
- helpers functions
- static functions

## Added
- Compatibility with PHP ^8.0
- Constructor
- Private properties
- Added getWith(), getHeight(), gethHash(), displayThumbhash()
- Moved extract_size_and_pixels_with_gd() and extract_size_and_pixels_with_imagick() from helpers to Thumbhash object

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-  "name": "srwiez/thumbhash",
+  "name": "jerrywham/thumbhash",
   "description": "A brief description of your package",
   "type": "library",
   "license": "MIT",
   "authors": [
     {
       "name": "Eser DENIZ",
-      "email": "srwiez@gmail.com"
+      "email": "srwiez@gmail.com",
```
Comment (mehdichaouch) [inline composer.json:9]: Correct format for this node is:

```json
  "authors": [
    {
      "name": "Eser DENIZ",
      "email": "srwiez@gmail.com"
    },{
      "name": "Cyril MAGUIRE",
      "email": "github@ecyseo.net"
    }
]
```

File: composer.json [hunk 2]
```diff
@@ -1,19 +1,20 @@
 {
-  "name": "srwiez/thumbhash",
+  "name": "jerrywham/thumbhash",
```
Comment (mehdichaouch) [inline composer.json:2]: This change will break the `composer require srwiez/thumbhash` command

File: readme.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ Thumbhash is a very compact representation of a placeholder for an image. Store
 You can install Thumbhash PHP using Composer:
 
 ```bash
-composer require srwiez/thumbhash
+composer require jerrywham/thumbhash
```
Comment (mehdichaouch) [inline readme.md:16]: same as in composer

File: src/Thumbhash.php [hunk 4]
```diff
@@ -2,8 +2,101 @@
...
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_gd($this->content);
+        }
+
+        $this->hash = $this->RGBAToHash($this->width, $this->height, $this->pixels);
+        $this->key = $this->convertHashToString($this->hash); // You can store this in your database as a string
+    }
+
+
+
+    private function extract_size_and_pixels_with_gd(string $content): array
```
Comment (SRWieZ) [inline src/Thumbhash.php:32]: Shouldn't be in this class. This code has nothing to do with thumbhash algorithm.

File: src/Thumbhash.php [hunk 5]
```diff
@@ -2,8 +2,101 @@
...
+        // $height = round(100 * $height / $size);
+
+        return [$width, $height, $pixels];
+    }
+
+    /**
+     * @throws \ImagickException
+     * @throws \ImagickPixelException
+     */
+    private function extract_size_and_pixels_with_imagick(string $content): array
```
Comment (SRWieZ) [inline src/Thumbhash.php:62]: Same here, nothing to do with thumbhash algorithm.

File: src/Thumbhash.php [hunk 6]
```diff
@@ -2,8 +2,101 @@
...
+        $this->content = file_get_contents($file);
+        if (class_exists('Imagick')) {
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_imagick($this->content);
+        } else {
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_gd($this->content);
+        }
+
+        $this->hash = $this->RGBAToHash($this->width, $this->height, $this->pixels);
+        $this->key = $this->convertHashToString($this->hash); // You can store this in your database as a string
+    }
```
Comment (SRWieZ) [inline src/Thumbhash.php:28]: I'm not against a real object but this constructor is wrong for a few reasons

1. Lot of framework uses [Flysystem](https://flysystem.thephpleague.com/docs/ ) abstraction to interact with files across different filesystems. [Laravel](https://laravel.com/docs/10.x/filesystem) even add a layer on top of that. I can also go further, what if my file comes from a stream ? This library shouldn't be responsible of file processing.
2. Actually this class is "unidirectional", we can't take base64 hash and decode the image.

**Review by SRWieZ — CHANGES_REQUESTED** [review]
I'm not against a real object but I don't think this is the right way of doing it.
See my comments for further explanations.
Also, I second @mehdichaouch comments.

Thank you for your time but I won't merge this class as it is. 
Feel free to come with a different approach if you feel to.

Have a good day

**Pull request closed without merging**